crates/sui-core/src/checkpoints/checkpoint_executor/mod.rs

```diff
@@ -289,15 +206,22 @@ impl CheckpointExecutorEventLoop {
                                 .unwrap();
                         }
                         Some(committee) => {
-                            debug!(
-                                "Last checkpoint ({:?}) of epoch {:?} has finished execution",
-                                checkpoint.sequence_number(),
-                                checkpoint.epoch(),
+                            info!(
+                                ended_epoch = checkpoint.epoch(),
+                                last_checkpoint=checkpoint.sequence_number(),
+                                "Reached end of epoch",
                             );
                             self.checkpoint_store
                                 .update_highest_executed_checkpoint(&checkpoint)
                                 .unwrap();
-                            return Some((checkpoint, committee));
+
+                            // be extra careful to ensure we don't have orphans
+                            assert!(
+                                pending.is_empty(),
+                                "Pending checkpoint execution buffer should be empty after processing last checkpoint of epoch",
+                            );
+                            drop(pending);
```

Comment:
you don't need to explicitly drop pending, it'll get dropped when it falls out of scope.

Reply:
Oh yeah not sure why I added this.